lib/command.js

```diff
@@ -931,12 +1012,124 @@ Expecting one of '${allowedValues.join("', '")}'`);
    */
 
   async parseAsync(argv, parseOptions) {
-    const userArgs = this._prepareUserArgs(argv, parseOptions);
-    await this._parseCommand([], userArgs);
+    this._asyncParsing = true;
+
+    try {
+      const userArgs = this._prepareUserArgs(argv, parseOptions);
+      await this._parseCommand([], userArgs);
+
+      return this;
+    } finally {
+      this._asyncParsing = undefined;
+    }
+  }
 
+  /**
+   * When set to `true`, thenable option and argument values will be awaited right after they are parsed and processed.
+   * When set to `undefined` (the default), inherit the behaviour from ancestor commands, or only await when `.parseAsync()` has been called.
+   * Useful for asynchronous custom processing (`parseArg`) of arguments and option-arguments.
+   *
+   * @param {boolean | undefined} [enabled]
+   * @return {Command} `this` command for chaining
+   */
+  await(enabled) {
+    this._await = enabled === undefined ? undefined : !!enabled;
     return this;
   }
 
+  /**
+   * If `_await` is `undefined`, inherit await behaviour or base it on whether `.parseAsync()` has been called on the top-level command.
+   *
+   * @api private
+   */
+
+  _shouldAwait() {
+    let shouldAwait;
+    let cmd = this;
+    do {
+      shouldAwait = cmd._await ?? cmd._asyncParsing;
```

Comment:
Here, `TypeError: Cannot read properties of null (reading '_await')` is thrown when `parse()` / `parseAsync()` is called on a command that does not "own" the subcommand anymore since it has been added as a subcommand to another parent command like in #1916.

The fact the error is thrown indicates
- either the wrong use of the library, in which case the while condition should be changed to `shouldAwait === undefined && cmd` and an error with a clear explanation of what has been done wrong should be thrown if `shouldAwait` is still `undefined` after the loop;
- or problems in the library's logic, in which case the suggestions from #1916 should be implemented.